mkosi/__init__.py

```diff
@@ -7635,8 +7643,16 @@ def find_address(args: MkosiArgs) -> Tuple[str, str]:
 
     die("Container/VM address not found")
 
+def run_command_image(args: MkosiArgs, commands: List[str], stdout: _FILE = sys.stdout, stderr: _FILE = sys.stderr) -> CompletedProcess:
+    if args.verb == "qemu":
+        return run_ssh(args, commands, stdout, stderr)
+    else:
+        cmdline = ['systemd-run', "--quiet", "--wait", "--pipe", "-M", virt_name(args)] + commands
+        with suppress_stacktrace():
+            return run(cmdline, stdout=stdout, stderr=stderr)
```

Comment:
I think you can add `text=True` here and in `run_ssh()` so that in the test you don't need to decode the output to UTF-8.